bundles/com.amazonaws.eclipse.core/src/com/amazonaws/eclipse/core/mobileanalytics/internal/ToolkitAnalyticsManagerImpl.java

```diff
@@ -148,4 +151,42 @@ public void setEnabled(boolean enabled) {
         this.enabled = enabled;
     }
 
+    private void injectAccountMetadata(MetricDatum metricDatum) {
+        final String awsAccount = userId;
+        String region = null;
+        try {
+            Region r = RegionUtils.getCurrentRegion();
+            if (r != null) {
+                region = r.getId();
+            }
+            // regionutils throws a runtime exception if it can't determine region, ignore
+            // if this happens
+        } catch (Exception e) {
+            ;
+        }
+        if (region != null && !region.isEmpty()) {
+            metricDatum.getMetadata().add(new MetadataEntry().key("awsRegion").value(region));
+        }
+        if (awsAccount != null && !awsAccount.isEmpty()) {
+            metricDatum.getMetadata().add(new MetadataEntry().key("awsAccount").value(userId));
+        }
+    }
+
+    private void getAccountIdFromSTS() {
```

Comment:
Is there a chance this gets called if the user doesn't have credentials set? If so, would we want to differentiate any errors from that from an error with the call itself?

Reply:
That would be fine, this is strictly for telemetry and we don't need 5 different types of not set like we have.

Reply:
`invalid` would be appropriate in error situations.

Reply:
We don't record this in any other IDE, I'd rather not make up something up for this now because we aready have `n/a` `not-set` and just not present at all

Reply:
I don't have strong opinions on this. 
[Here](https://github.com/aws/aws-toolkit-vscode/blob/master/src/shared/telemetry/defaultTelemetryService.ts#L220) is what VS Code Toolkit is doing.